cube_helper/fix_known.py

```diff
@@ -0,0 +1,180 @@
+        self.cube = cube
+        self.logger = log_module()
+
+    @abstractmethod
+    def is_fix_needed(self):
+        """
+        Check if this fix applies to the object's cube.
+
+        Returns:
+            True if this fix should be applied to the cube that it was
+            instantiated with.
+        """
+        pass
+
+    @abstractmethod
+    def fix_cube(self):
+        """
+        Apply the fix to the object's cube and warn about the name of the fix
+        that has been applied.
+        """
+        log_message = 'Applying {}'.format(self.__class__.__name__)
+        self.logger.info(log_message)
+
+
+@six.add_metaclass(ABCMeta)
+class FixKnownIssueIdentifyAttributes(FixKnownIssue):
+    """
+    An abstract base class to identify and fix known issues with data loaded
+    into an `iris.cube.Cube`. A cube that needs this fix is identified by
+    the class' `cube_attributes_required` attribute key-value pairs being
+    found in `cube.attributes`.
+
+    Args:
+        cube: the cube to check, and if necessary fix.
+    """
+    @abstractmethod
+    def __init__(self, cube):
+        super(FixKnownIssueIdentifyAttributes, self).__init__(cube)
+        self.cube_attributes_required = {}
+
+    def is_fix_needed(self):
+        """
+        Check if this fix applies to the object's cube by checking that all of
+        the key-value pairs in `self.cube_attributes_required` occur in
+        `cube.attributes`.
+
+        Returns:
+            True if this fix should be applied to the cube that it was
+            instantiated with.
+        """
+        for key, value in six.iteritems(self.cube_attributes_required):
+            if key in self.cube.attributes:
+                if self.cube.attributes[key] != value:
+                    return False
+            else:
+                return False
+        # Haven't found anything that doesn't match
+        return True
+
+
+class FixCmip6CasFgoals(FixKnownIssueIdentifyAttributes):
+    """
+    Fix the latitude and longitude bounds in mip_era: CMIP6 institution_id: CAS
+    source_id: FGOALS-f3-L, which are not contiguous and use
+    `iris.coords.Coord.guess_bounds()` to calculate new bounds for these two
+    coordinates.
+
+    Args:
+        cube: the cube to check, and if necessary fix.
+    """
+    def __init__(self, cube):
+        super(FixCmip6CasFgoals, self).__init__(cube)
+        self.cube_attributes_required = {
+            'mip_era': 'CMIP6',
+            'institution_id': 'CAS',
+            'source_id': 'FGOALS-f3-L'
+        }
+
+    def fix_cube(self):
+        """
+        Delete the existing faulty bounds on the latitude and longitude
+        coordinates and then calculate new bounds.
+        """
+        super(FixCmip6CasFgoals, self).fix_cube()
+        self.cube.coord('latitude').bounds = None
+        self.cube.coord('latitude').guess_bounds()
+        self.cube.coord('longitude').bounds = None
+        self.cube.coord('longitude').guess_bounds()
```

Comment:
Will these coordinates always be called `latitude` and `longitude`? Would it be worth using `axis='Y'` and `axis='X'`, see https://scitools.org.uk/iris/docs/latest/iris/iris/cube.html#iris.cube.Cube.coord?